TedTalkMVVM/TedTalkMVVM/view/HomeViewController.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  ViewController.swift
+//  TedTalkMVVM
+//
+//  Created by Gonzalo Romero on 09/12/2021.
+//
+
+import UIKit
+
+class HomeViewController: UIViewController {
+    
+    @IBOutlet weak var tableTedTalk: UITableView!
+    @IBOutlet weak var pickerTedTalk: UIPickerView!
+    @IBOutlet weak var searchTedTalk: UISearchBar!
+    
+    var viewModel: ViewModelTedTalkProtocol!
+    var tedTalksDisplay: [TedTalkDisplay] = []
+    private var pickerSelectedRow = ""
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        viewModel = ViewModelTedTalk(displayManager: DisplayManagerTedTalk.shareTedTalkManager ,observerData: { [unowned self] viewModel in
+            self.tedTalksDisplay = viewModel
+            self.reloadData()
+        })
+    }
+    
+    func reloadData() {
+        DispatchQueue.main.async {
+            self.tableTedTalk.reloadData()    }
+    }
+}
+extension HomeViewController: UITableViewDataSource, UISearchBarDelegate, UIPickerViewDataSource, UIPickerViewDelegate, UITableViewDelegate{
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        viewModel.getPickerRowsCount()
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        viewModel.getPickerRow(index: row)
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        pickerSelectedRow = viewModel.getPickerRow(index: row)
+    }
```

Comment:
instead of asking to the viewModel to get picker data you should implement an approach similary as you did for the table view